libraries/engage/core/hooks/usePageSettings.js

```diff
@@ -1,11 +1,14 @@
-import { usePageConfig } from './usePageConfig';
+import { useRoute } from './useRoute';
+import { getPageSettings } from '../config/getPageSettings';
 
 /**
  * Retrieves the settings for the current page.
+ * See documentation of `getPageSettings` for further details.
+ *
+ * @param {string} [key=null] Optional key to read and inherit from theme settings
  * @returns {Object}
  */
-export function usePageSettings() {
-  const { settings = {} } = usePageConfig();
-
-  return settings;
+export function usePageSettings(key = null) {
+  const { pattern: pagePattern } = useRoute();
```

Comment:
This is not needed if the `useCurrentRoute()` helper is used inside `getPageConfig()` as mentioned above.

Reply:
See my previous comment.